cli/plasmo/src/features/manifest-factory/base.ts

```diff
@@ -419,17 +420,47 @@ export abstract class PlasmoManifest<T extends ExtensionManifest = any> {
       .then((files) =>
         Promise.all(
           files
-            .filter((f) => f.isFile())
+            .filter((f) =>
+              f.isFile() && filterFile ? filterFile(f.name) : true
+            )
             .map((f) => resolve(path, f.name))
             .map((filePath) => toggleDynamicPath(filePath, true))
         )
       )
       .then((results) => results.includes(true))
   }
 
+  addContentScriptsNestedDirectories = async () => {
```

Comment:
This could be a bit more generic and accept a `path`, though I kept it a bit more narrow since (afaik) content scripts are the only case currently which needs this logic